File: frontend/src/types/borrowTransaction.ts
export interface BorrowTransaction {
  transaction_id: string;
  book_id: string;
  user_id: string;
  borrowed_at: string;
  due_date: string;
  status: string;
  returned_at?: string;
}

export interface BorrowBookResponse {
  transaction_id: string;
  book_id: string;
  user_id: string;
  borrowed_at: string;
  due_date: string;
  status: string;
}

File: backend/app/schemas/borrow_transaction.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class BorrowTransactionResponse(BaseModel):
    id: str
    book_id: str
    user_id: str
    borrowed_at: datetime
    due_date: datetime
    status: str
    returned_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class BorrowBookResponse(BaseModel):
    transaction_id: str
    book_id: str
    user_id: str
    borrowed_at: datetime
    due_date: datetime
    status: str

    model_config = {"from_attributes": True}
